### ISSUE TITLE
Pooling layer issue with CPP parser 

### ISSUE THREAD

**Issue by AnthonyARM** [KhronosGroup/NNEF-Tools#55]
If I try to load `examples/googlenet_comp.txt` with the CPP parser and I don't support the `max_pool_layer` and `avg_pool_layer` then it falls back onto calling `max_pool` and `avg_pool` however the parameters passed don't match what the spec suggests:
```cpp
pool5 = avg_pool_layer(incept9, size = [7,7]);
```
Gets translated into:
```cpp
pool5 = avg_pool
Pooling args : {border: 'constant', dilation: [], input: incept9, output: pool5, padding: [], size: [1,1,7,7], stride: []}
```
Whereas the spec says:
```
fragment avg_pool_layer(
    input: tensor<scalar>,
    size: integer[],
    border: string = 'constant',
    padding: (integer,integer)[] = [],
    stride: integer[] = [],
    dilation: integer[] = [] )
-> ( output: tensor<scalar> )
{
    output = avg_pool(input, size = [1,1] + size, border = border,
                      padding = [(0,0), (0,0)] + padding
                                if length_of(padding) != 0 else [],
                      stride = [1,1] + stride
                                if length_of(stride) != 0 else [],
                      dilation = [1,1] + dilation
                                if length_of(dilation) != 0 else []);
}
```
So, I would expect to receive:
```
Pooling args : {border: 'constant', dilation: [1,1], input: incept9, output: pool5, padding: [(0,0),(0,0)], size: [1,1,7,7], stride: [1,1]}
```
No ?

**Comment by AnthonyARM** [comment]
Well, actually, looking a bit more into it, maybe the problem comes from the example:
```cpp
    pool5 = avg_pool_layer(incept9, size = [7,7]);
    //  Instantiated pool5 Type: PoolingLayer Data Type: F32 Input shape: 7x7x1024 Output shape: 7x7x1024 Pooling info: {Type=AVG,IsGlobalPooling=0,PoolSize=7,7,PadStride=1,1;3,3,3,3}

    conv4 = conv_layer(pool5, channels = 1000, size = [1,1], scope = 'conv4');
    //Instantiated ConvolutionLayer Data Type: F32 Input shape: 7x7x1024 Weights shape: 1x1x1024x1000 Output shape: 7x7x1000
    logits = softmax(conv4);
    //Instantiated logits Type: SoftmaxLayer Data Type: F32 Input shape: 7x7x1000 Output shape: 7x7x1000
```

I think the problem comes from the `avg_pool_layer`, shouldn't the padding be forced to `[(0,0),(0,0)]` in order to go from 7x7x1024 to 1x1x1024 ?
If I set it to 
```
pool5 = avg_pool_layer(incept9, size = [7,7], stride=[1,1], padding = [(0,0), (0,0)]);
```
Then logits becomes `1x1x1000` as you would expect.

**Comment by AnthonyARM** [comment]
Oh ok, I guess it makes sense. Closing then.
Thanks for clarifying.
